testing/simapp/app.go

```diff
@@ -245,6 +245,7 @@ type SimApp struct {
 
 	// make IBC modules public for test purposes
 	// these modules are never directly routed to by the IBC Router
+	IBCMockModule ibcmock.IBCModule
```

Comment:
Side question, how come this wasn't present earlier?

Reply:
I believe we originally added these overrides to test middleware. One initial point of testing was with interchain accounts (to mimic "icauth" modules, v6 improvement removed notion of "icaauth"). The mock module would be wrapped however as it acts as middleware, thus you reference `ICAAuthModule` instead of `IBCMockModule`

Referencing the unwrapped mock module is only required when you don't need to test the mock module as middleware but as a base application. This only occurs when you need to revert state of the IBC app callback. This occurs in `OnRecvPacket` handling. Because those tests were written before this feature was added, it uses a [`mock.MockFailPacketData`](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server_test.go#L77). Then logic was added to `mock.OnRecvPacket` to return a certain ack based on the [packet data](https://github.com/cosmos/ibc-go/blob/main/testing/mock/ibc_module.go#L124). These tests could be updated to use the override app callback. It'd probably be more stable in the long term (as you won't need to pass in a certain packet data to create a certain result)

Reply:
#432

Reply:
looks like I initially added `GetMockModule()` but I'm not sure when that was removed

Reply:
It was removed in this pr https://github.com/cosmos/ibc-go/pull/380, but idk why. Might have been something as simple as a merge conflict mistake or unused func

Reply:
Yeah, must have been a merge conflict I assume. I was a little surprised when I noticed this was missing from the SimApp struct.